File: spring-cloud-stream/rabbit-consumer/src/main/java/com/livk/consumer/RabbitConsumer.java
package com.livk.consumer;


import com.livk.spring.LivkSpring;
import com.livk.stream.entity.StreamMessage;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;
import reactor.core.publisher.Flux;

import java.util.function.Consumer;

/**
 * <p>
 * KafkaConsumer
 * </p>
 *
 * @author livk
 * @date 2022/2/15
 */
@Slf4j
@SpringBootApplication
public class RabbitConsumer {

    public static void main(String[] args) {
        LivkSpring.run(RabbitConsumer.class, args);
    }

    @Bean
    public Consumer<Flux<StreamMessage<String>>> send() {
        return streamMessageFlux -> streamMessageFlux.subscribe(streamMessage -> log.info("[{}]", streamMessage));
    }

}

File: spring-cloud-stream/rabbit-consumer/src/main/kotlin/com/livk/consumer/RabbitConsumer.kt
package com.livk.consumer

import com.livk.stream.entity.StreamMessage
import org.slf4j.LoggerFactory
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Bean
import reactor.core.publisher.Flux

/**
 * @author livk
 */
@SpringBootApplication
open class RabbitConsumer {

    private val log = LoggerFactory.getLogger(RabbitConsumer::class.java)

    @Bean
    open fun send(): (Flux<StreamMessage<String>>) -> Unit = { streamMessageFlux ->
        streamMessageFlux.subscribe { streamMessage ->
            log.info(
                "[{}]",
                streamMessage
            )
        }
    }
}

fun main(args: Array<String>) {
    runApplication<RabbitConsumer>(*args)
}
